File: module-persistence/src/main/java/com/hanghae/module/persistence/repository/impl/ScreeningRepositoryImpl.java
package com.hanghae.module.persistence.repository.impl;

import com.hanghae.module.domain.model.Screening;
import com.hanghae.module.domain.repository.ScreeningRepository;
import com.hanghae.module.persistence.entity.ScreeningEntity;
import com.hanghae.module.persistence.mapper.ScreeningMapper;
import com.hanghae.module.persistence.repository.jpa.ScreeningJpaRepository;
import jakarta.persistence.EntityNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class ScreeningRepositoryImpl implements ScreeningRepository {
  private final ScreeningJpaRepository screeningJpaRepository;
  private final ScreeningMapper screeningMapper;
  @Override
  public Screening findById(Long id) {
    ScreeningEntity entity = screeningJpaRepository.findById(id)
      .orElseThrow(() -> new EntityNotFoundException("존재하지 않는 상영 정보입니다."));
    return screeningMapper.toDomain(entity);
  }
}

File: infrastructure/mysql/src/main/kotlin/com/hanghe/redis/screening/ScreeningRepositoryImpl.kt
package com.hanghe.redis.screening

import jakarta.persistence.EntityNotFoundException
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Repository

@Repository
class ScreeningRepositoryImpl(
    private val screeningJpaRepository: ScreeningJpaRepository
) : ScreeningRepository {

    override fun findByMovieIdOrderByStartTime(movieId: Long): List<ScreeningEntity> {
        return screeningJpaRepository.findByMovieIdOrderByStartTime(movieId)
    }

    override fun saveAll(movies: List<ScreeningEntity>): List<ScreeningEntity> {
        return screeningJpaRepository.saveAll(movies)
    }

    override fun findById(screeningId: Long): ScreeningEntity? {
        return screeningJpaRepository.findByIdOrNull(screeningId)
    }

    override fun getById(screeningId: Long): ScreeningEntity {
        return findById(screeningId) ?: throw EntityNotFoundException()
    }
}
